contracts/clients/src/BlsSigner.ts

```diff
@@ -200,17 +200,56 @@ export default class BlsSigner extends Signer {
     throw new Error("_signTypedData() is not implemented.");
   }
 
-  connectUnchecked(): ethers.providers.JsonRpcSigner {
-    throw new Error("connectUnchecked() is not implemented.");
+  connectUnchecked(): BlsSigner {
+    throw new Error(
+      "connectUnchecked() is not implemented. Use connectBlsUnchecked instead.",
+    );
+  }
+
+  // When we use regular "connectUnchecked", we need to pass the private key into the method as that is a required
+  // argument for the BlsSigner constructor. This breaks liskov substitution as the subclass method signature is different.
+  connectBlsUnchecked(privateKey: string): BlsSigner {
+    return new UncheckedBlsSigner(
+      _constructorGuard,
+      this.provider,
+      privateKey,
+      this._address || this._index,
+    );
```

Comment:
```suggestion
  connectUnchecked(): BlsSigner {
    return new UncheckedBlsSigner(
      _constructorGuard,
      this.provider,
      // Modify this arg to take a priv key OR promise that will eventually resolve to one, then do similar initPromise guards in this class.
      this.wallet?.privateKey ?? async () => {
      	await this.initPromise();
      	return this.wallet.privateKey;
      },
      this._address || this._index,
    );
  }
```

Reply:
This requires modifying the `BlsSigner` constructor to accept a `privateKey` or a promise that will eventually resolve to one. We can discuss offline some cleaner ways to do this.